src/iop/colorbalance.c

```diff
@@ -1608,7 +1610,7 @@ static void gain_neutralize_callback(GtkWidget *button, gpointer user_data)
     
     // Save the patch color for the optimization
     dt_iop_colorbalance_data_t *d = (dt_iop_colorbalance_data_t *)self->data;
-    *d->color_patches_gain = *RGB;
+    for (int c = 0; c < 3; c++) d->color_patches_gain[c] = RGB[c];
```

Comment:
Is that still correct?

I mean you're mapping RGB to  CHANNEL_FACTOR, CHANNEL_RED, CHANNEL_GREEN.

Maybe: d->color_patches_gain[c+1] = RGB[c];

But this means CHANNEL_FACTOR is not set.

Reply:
Actually the code only uses the first 3 cells of the array, from 0, so the tags name don't matter much.

Line 1671: 

```
  for (int c = 0; c < 3; ++c) 
  {
    samples_lift[c] = CDL(d->color_patches_lift[c], p->gain[CHANNEL_FACTOR], p->lift[CHANNEL_FACTOR] - 1.0f, 1.0f / p->gamma[CHANNEL_FACTOR]);
    samples_gamma[c] = CDL(d->color_patches_gamma[c], p->gain[CHANNEL_FACTOR], p->lift[CHANNEL_FACTOR] - 1.0f, 1.0f / p->gamma[CHANNEL_FACTOR]);
    samples_gain[c] = CDL(d->color_patches_gain[c], p->gain[CHANNEL_FACTOR], p->lift[CHANNEL_FACTOR] - 1.0f, 1.0f / p->gamma[CHANNEL_FACTOR]);
  }
```